src/main.js

```diff
@@ -153,44 +153,102 @@ function getEstimatedPaymentForTimeInUSD(timeSpent) {
 }
 
 /***********************************************************
- * Payment Streamed Calculation
+ * Amount of Payment Streamed (Sent Assets)
  ***********************************************************/
 
-/**
- * Calculate the total sent XRP across all origins.
- *
- * @return {Number} The total sent XRP across all origins.
- */
-async function calculateTotalSentXRP() {
-	const originDataList = await getOriginDataList();
-	let totalSentXRP = 0;
+function getTotalSentAssets(originStats) {
+	if (originStats.totalSentAssetsMap) {
+		let sentAssetsMap = new Map();
 
-	for (const originData in originDataList) {
-		totalSentXRP += calculateTotalSentXRPForOrigin(originData);
+		for (const sentAssetData of Object.values(originStats.totalSentAssetsMap)) {
+			const currency = sentAssetData.assetCode;
+			let amount = sentAssetData.amount * (10**(-sentAssetData.assetScale));
+			sentAssetsMap.set(currency, amount);
+		}
+
+		if (sentAssetsMap.size > 0) {
+			return sentAssetsMap;
+		}
 	}
 
-	return unNaN(totalSentXRP);
+	return null;
 }
 
 /**
- * Calculate the total sent XRP for the specified origin.
+ * Calculate the total sent assets at the origin for each currency sent to the payment
+ * pointers seen at the origin and return the data as a map.
  *
- * @param {AkitaOriginData} originData The AkitaOriginData to calculate sent XRP for.
- * @return {Number} The total sent XRP for the specified origin.
+ * @param {AkitaOriginData} originData The AkitaOriginData to calculate sent assets for.
+ * @return {Map<String, Number>} A map containing the sent asset amounts, with the currency
+ * as the key (String) and the sent amount as the value (Number).
  */
-function calculateTotalSentXRPForOrigin(originData) {
-	let totalSentXRP = 0;
-
+function getTotalSentAssetsForOrigin(originData) {
 	if (originData.paymentPointerMap) {
+		let sentAssetsMap = new Map();
+
 		for (const paymentPointerData of Object.values(originData.paymentPointerMap)) {
-			if (paymentPointerData.sentAssetsMap?.XRP?.amount > 0) {
-				const sentXRP = paymentPointerData.sentAssetsMap.XRP;
-				const actualAmount = sentXRP.amount * (10**(-sentXRP.assetScale));
-				totalSentXRP += actualAmount;
+			for (const sentAssetData of Object.values(paymentPointerData.sentAssetsMap)) {
+				const currency = sentAssetData.assetCode;
+				let amount = sentAssetData.amount * (10**(-sentAssetData.assetScale));
+				const existingAmountForCurrency = sentAssetsMap.get(currency);
```

Comment:
nit: move this inside the following `if` statement since it's only used inside there.